client/src/components/SubNav.js

```diff
@@ -12,17 +12,25 @@ export default class SubNav extends Component {
 		super(props)
 
 		this.state = {
-			subnavElem: document.getElementById(this.props.subnavElemId),
+			subnavElem: null,
 		}
 	}
 
-	componentDidMount() {
+	updateSubnav = () => {
 		const elem = document.getElementById(this.props.subnavElemId)
```

Comment:
Can we use https://reactjs.org/docs/refs-and-the-dom.html instead of `getElementById`?

Reply:
According to https://reactjs.org/docs/portals.html the second arg to `createPortal` must be a DOM node.

Reply:
`.current`?

Reply:
Re: your first remark; changing it to just this (plus `propTypes`):
```
	render() {
		return (
			ReactDOM.createPortal(
				this.props.children,
				document.getElementById(this.props.subnavElemId)
			)
		)
	}
```
could probably work.

Reply:
But I get an occasional `Error: Target container is not a DOM element.` when trying that.

Reply:
```
	render() {
		const subnavElem = document.getElementById(this.props.subnavElemId)
		return subnavElem && ReactDOM.createPortal(
			this.props.children,
			subnavElem
		)
	}
```
works much better!